Fix missing num_hidden_layers attribute in T5GemmaConfig

0xjeffro opened the pull request:
This is a quick fix for issue #40874. The `T5GemmaConfig` class was missing the `num_hidden_layers` attribute that cache initialization expects. Added `num_hidden_layers` property to expose the decoder's `num_hidden_layers` value.

# What does this PR do?

<!--
Congratulations! You've made it this far! You're not quite done yet though.

Once merged, your PR is going to appear in the release notes with the title you set, so make sure it's a great title that fully reflects the extent of your awesome contribution.

Then, please replace this with a description of the change and which issue is fixed (if applicable). Please also include relevant motivation and context. List any dependencies (if any) that are required for this change.

Once you're done, someone will review your PR shortly (see the section "Who can review?" below to tag some potential reviewers). They may suggest changes to make the code even better. If no one reviewed your PR after a week has passed, don't hesitate to post a new comment @-mentioning the same persons---sometimes notifications get lost.
-->

<!-- Remove if not applicable -->

Fixes #40874 


## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?


## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

<!-- Your PR will be replied to more quickly if you can figure out the right person to tag with @

 If you know how to use git blame, that is the easiest way, otherwise, here is a rough guide of **who to tag**.
 Please tag fewer than 3 people.

Models:

- text models: @ArthurZucker
- vision models: @amyeroberts, @qubvel
- speech models: @eustlb
- graph models: @clefourrier

Library:

- flax: @gante and @Rocketknight1
- generate: @zucchini-nlp (visual-language models) or @gante (all others)
- pipelines: @Rocketknight1
- tensorflow: @gante and @Rocketknight1
- tokenizers: @ArthurZucker
- trainer: @zach-huggingface, @SunMarc and @qgallouedec
- chat templates: @Rocketknight1

Integrations:

- deepspeed: HF Trainer/Accelerate: @SunMarc @zach-huggingface
- ray/raytune: @richardliaw, @amogkam
- Big Model Inference: @SunMarc
- quantization (bitsandbytes, autogpt): @SunMarc @MekkCyber

Documentation: @stevhliu

HF projects:

- accelerate: [different repo](https://github.com/huggingface/accelerate)
- datasets: [different repo](https://github.com/huggingface/datasets)
- diffusers: [different repo](https://github.com/huggingface/diffusers)
- rust tokenizers: [different repo](https://github.com/huggingface/tokenizers)

Maintained examples (not research project or legacy):

- Flax: @Rocketknight1
- PyTorch: See Models above and tag the person corresponding to the modality of the example.
- TensorFlow: @Rocketknight1

 -->

0xjeffro committed:
Fix missing num_hidden_layers attribute in T5GemmaConfig

This is a quick fix for issue #40874. The `T5GemmaConfig.get_text_config()` method was always returning self instead of the appropriate sub-configuration. This caused cache initialization failures because the main config lacks `num_hidden_layers` while the decoder/encoder sub-configs have this attribute.

Now the method correctly returns the decoder config when `decoder=True` and encoder config when `encoder=True`, allowing `cache_utils.py` to obtain the proper configuration with `num_hidden_layers` for cache initialization.

0xjeffro committed:
Fix missing num_hidden_layers attribute in T5GemmaConfig

This is a quick fix for issue #40874. The `T5GemmaConfig` class was missing the `num_hidden_layers` attribute that cache initialization expects. Added `attribute_map` and `num_layers` property following the same pattern used by `T5Config` to expose the decoder's `num_hidden_layers` value.

0xjeffro committed:
Fix missing num_hidden_layers attribute in T5GemmaConfig

This is a quick fix for issue #40874. The `T5GemmaConfig` class was missing the `num_hidden_layers` attribute that cache initialization expects. Added `attribute_map` and `num_layers` property following the same pattern used by `T5Config` to expose the decoder's `num_hidden_layers` value.

0xjeffro committed:
Merge branch 'main' into fix_missing_num_hidden_layers_in_T5GemmaConfig

ArthurZucker commented on src/transformers/models/t5gemma/modular_t5gemma.py:
```diff
@@ -100,6 +100,7 @@ class T5GemmaConfig(PretrainedConfig):
 
     model_type = "t5gemma"
     keys_to_ignore_at_inference = ["past_key_values"]
+    attribute_map = {"num_hidden_layers": "num_layers"}
```
why not directly declare self.num_hidden_layers? 😉

ArthurZucker reviewed (commented):
forgot that it was an encoder decoder hahah there might be a better way to fix this @gante I don't remember what we do in that case

0xjeffro committed:
Merge remote-tracking branch 'origin/fix_missing_num_hidden_layers_in_T5GemmaConfig' into fix_missing_num_hidden_layers_in_T5GemmaConfig

# Conflicts:
#	src/transformers/models/t5gemma/configuration_t5gemma.py

0xjeffro committed:
Fix missing num_hidden_layers attribute in T5GemmaConfig

This is a quick fix for issue #40874. The `T5GemmaConfig` class was missing the `num_hidden_layers` attribute that cache initialization expects. Added `num_hidden_layers` property to expose the decoder's `num_hidden_layers` value.

0xjeffro committed:
Merge branch 'main' into fix_missing_num_hidden_layers_in_T5GemmaConfig

0xjeffro committed:
Fix missing num_hidden_layers attribute in T5GemmaConfig

This is a quick fix for issue #40874. The `T5GemmaConfig` class was missing the `num_hidden_layers` attribute that cache initialization expects. Added `num_hidden_layers` property to expose the decoder's `num_hidden_layers` value.

0xjeffro committed:
Merge branch 'main' into fix_missing_num_hidden_layers_in_T5GemmaConfig

0xjeffro committed:
Merge branch 'main' into fix_missing_num_hidden_layers_in_T5GemmaConfig

gante reviewed (changes requested):
uhmmm this is not wanted, it means `config.get_text_config` is not fetching the decoder for this model correctly. Having a look at the original issue

gante commented:
#40939 is the right fix 🤗
